src/utils.ts

```diff
@@ -27,19 +27,28 @@ interface AnyObject {
   [key: string]: any;
 }
 
+function transformArrayToObject(source: AnyObject) {
+  if (Array.isArray(source) && source.map(s => typeof s === 'object').reduce((a, b) => a && b, true)) {
+    return source.reduce((a, b) => Object.assign(a, b), {})
+  }
+  return source
+}
+
 export function deepAssign(target: AnyObject, source: AnyObject): AnyObject {
-  Object.keys(source).forEach(k => {
+  const tt = transformArrayToObject(target)
```

Comment:
Thought it better to transform array of objects to a single object to allow matching by object keys rather that array keys ('0', '1', ...)